docs/guides/08-animate-your-visualizations.md

```diff
@@ -0,0 +1,410 @@
+const $currentTime = document.getElementById('js-current-time');
+```
+
+**Note:**
+This guide makes use of some conventions. When setting an `id` to an element that is going to be accessed via JavaScript, the `id` starts with `js`. In addition, when assigning the element (`const $progressBanner = document.getElementById('js-progress-banner')`), the JavaScript value starts with `$`, which indicates that it contains an HTML element.
+
+
+#### Add listening events
+In this step, we will add listening events tied to the different interaction buttons used to control the animation as described above. For example, we will tell the **Pause** button that it has to be ready and react when clicked to pause the animation.
+
+```js
+$pauseButton.addEventListener('click', () => {
+    viz.variables.animation.pause();
+});
+```
+
+And the same applies to the **Play** button:
+
+```js
+$playButton.addEventListener('click', () => {
+    viz.variables.animation.play();
+});
+```
+
+In the same way, we can update the **duration** of the animation when the duration range is adjusted in the slider:
+
+```js
+$durationRange.addEventListener('change', () => {
+    viz.variables.duration = parseInt($durationRange.value, 10);
+});
+```
+
+
+#### Update the progress bar
+Now that we have added the listening events for the different buttons and defined the behavior upon interaction, next we will walk through updating the progress bar and the current time text.
+
+The `updateProgress` function will be responsible for:
+- updating the progress bar through the duration of the animation, using the range value provided by the `getProgressPct()` method and
+- displaying the current time, by updating a span with the current `getProgressValue()` method:
+
+```js
+function updateProgress () {
+    $progressRange.value = viz.variables.animation.getProgressPct();
+    $currentTime.innerText = viz.variables.animation.getProgressValue();
+}
+```
+
+Let’s call this function periodically by using `setInterval`:
+
+```js
+setInterval(updateProgress, 100);
+```
+
+**Note:**
+The `setInterval()` is a commond method available at the browser, that calls a function or evaluates an expression at specified intervals, in milliseconds. You can also do this update by listening to `carto.layer` events. This topic is covered more in-depth in the [Interactivity and Events Guide](/developers/carto-vl/guides/interactivity-events/); once you know how, the second options is preferred.
+
+
+#### All together
+Well done! In the example below you will find the full code, including the animation controls. You can play with it and make adjustments.
+
+Remember you can style your HTML controls using CSS. These styles are very simple, but at the beginning of this guide you have a complete example that includes beautiful styles. This example is also available in the [examples section](/developers/carto-vl/examples/#example-animation-controls).
+
+<div class="example-map">
```

Comment:
Clicking on the progress bar is broken

Reply:
@davidmanzanares The progress bar in this example-5 has been kept "simple", not interactive, to avoid more code at this step. But we have that feature in example-7.

What I have found is a little bug, regarding to changing the animation's duration to 0 as it is running, I'll create the issue

Reply:
As that one gets solved, I'm gonna set the minimum duration in the slider to 1